jira/client.py

```diff
@@ -2905,7 +2906,7 @@ def search_assignable_users_for_issues(
         assignees, specify an issue key.
```

Comment:
Let's also add in the docstring why we have both `query` and `username` similar to the referenced pull request:

https://github.com/pycontribs/jira/pull/927/files#diff-442d14b20d591afd0c3de830d2803e30d4816eedc708f9ad129316aa7e5a4a7cR2970-R2971